app/src/main/java/org/oxycblt/auxio/music/Music.kt

```diff
@@ -366,50 +368,30 @@ class SortName(name: String, musicSettings: MusicSettings) : Comparable<SortName
                     }
                 }
 
-            // Parse out numeric portions of the title and use those for sorting, if applicable.
-            when (val numericEnd = sortName.indexOfFirst { !it.isDigit() }) {
-                // No numeric component.
-                0 -> number = null
-                // Whole title is numeric.
-                -1 -> {
-                    number = sortName.toIntOrNull()
-                    sortName = ""
-                }
-                // Part of the title is numeric.
-                else -> {
-                    number = sortName.slice(0 until numericEnd).toIntOrNull()
-                    sortName = sortName.slice(numericEnd until sortName.length)
-                }
-            }
-        } else {
-            number = null
+            // Zero pad all numbers to (an arbitrary) five digits for better sorting
+            sortName = sortName.replace(Regex("""\d+""")) { it.value.padStart(5, '0') }
```

Comment:
Why just five? This is asking for edge-cases to appear.

Reply:
Generally, I think I will need to devise a smarter algorithm for integer component handling that:

1. Handles all numerical components in a title
2. Can handle arbitrary number precision (No overflows or digit issues)
3. Relatively fast

I'd recommend just raising it to 6 for now as a stopgap measure.